.github/scripts/pre-commit.sh

```diff
@@ -74,5 +74,19 @@ runShellCheckForCommitFiles() {
     exit 1
   fi
 }
+
+checkGitleaks() {
+  echo "Attempting to execute gitleaks. This may take a minute..."
+  if ! command -v gitleaks >/dev/null; then
+    echo "'gitleaks' not found. Install gitleaks before pushing your changes."
+    return 1
+  fi
+  if gitleaks protect --staged --verbose; then
```

Comment:
While this `if` structure is explicit, it's not entirely necessary with the overarching `set -e`. `checkGitLeaks` _could_ be simplified to a single-line, pre-commit operation.